File: src/main/java/com/cout970/computer2/network/ChannelManager.java
package com.cout970.computer2.network;

import com.cout970.computer2.ComputerMod;
import net.darkaqua.blacksmith.api.common.network.INetworkChannel;
import net.darkaqua.blacksmith.api.common.network.NetworkChannelFactory;
import net.darkaqua.blacksmith.api.common.util.GameSide;

/**
 * Created by cout970 on 09/01/2016.
 */
public class ChannelManager {

    public static final INetworkChannel channel = NetworkChannelFactory.createNetworkChannel(ComputerMod.MOD_ID);

    public static void init(){
        channel.registerMessage(new MessageGenericUpdate(), MessageGenericUpdate.class, 0, GameSide.SERVER);
        channel.registerMessage(new MessageGenericUpdate(), MessageGenericUpdate.class, 1, GameSide.CLIENT);
        channel.registerMessage(new MessageButtonClick(), MessageButtonClick.class, 2, GameSide.SERVER);
    }

}

File: src/main/kotlin/com/cout970/computer/network/ChannelManager.kt
package com.cout970.computer.network

import com.cout970.computer.misc.MOD_ID
import net.minecraftforge.fml.common.network.simpleimpl.SimpleNetworkWrapper
import net.minecraftforge.fml.relauncher.Side

/**
 * Created by cout970 on 20/05/2016.
 */
object ChannelManager {

    val channel = SimpleNetworkWrapper(MOD_ID)

    init {
        channel.registerMessage(MessageGenericUpdate(), MessageGenericUpdate::class.java, 0, Side.SERVER)
        channel.registerMessage(MessageGenericUpdate(), MessageGenericUpdate::class.java, 1, Side.CLIENT)
        channel.registerMessage(MessageButtonClick(), MessageButtonClick::class.java, 2, Side.SERVER)
    }
}